lib/floe/workflow/intrinsic_function/transformer.rb

```diff
@@ -34,6 +34,22 @@ def self.resolve_args(args)
           Transformer.resolve_args(args)
         end
 
+        rule(:states_array_partition => {:args => subtree(:args)}) do
+          args = Transformer.resolve_args(args())
+          raise ArgumentError, "wrong number of arguments to States.ArrayPartition (given #{args.size}, expected 2)" unless args.size == 2
```

Comment:
As discussed in #238, going to leave this for now, and then I'll have a follow up PR using better exception types.